### PR TITLE
chore: Go releaser workflow

### LINKED ISSUE #470: Binary releases with goreleaser
## Problem

Currently the CI that produces binary releases is broken or it is not configured to push compiled binary as part of the releases. 

## What I'd like to see

Releases to have compiled binaries for the major platforms ( https://github.com/masa-finance/masa-oracle/releases) 

Go releaser can be used to speed-up this process and avoid wasting engineering cycles on designing a custom release process.

## Notes

https://goreleaser.com/

### Examples from other repositories

Goreleaser configuration file (that's a good starting point/example) https://github.com/mudler/edgevpn/blob/master/.goreleaser.yml
Github action workflow: https://github.com/mudler/edgevpn/blob/master/.github/workflows/release.yml

## Action items

- [ ] Have a GHA workflow that pushes compiled binaries during releases
- [x] Use goreleaser to build distributable tarballs
- [ ] the GHA workflow should use goreleaser

### PR DESCRIPTION
**Pull request by nolanjacobson**
**Description**

This PR fixes #470 

**Notes for Reviewers**

**[Signed commits](../CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

File: .github/workflows/.goreleaser.yml
Comment (mudler) [inline .github/workflows/.goreleaser.yml:37]: probably this is going to overlap with the automatic release notes generated by github - however didn't checked, I might be wrong here

File: .github/workflows/.goreleaser.yml
Comment (mudler) [inline .github/workflows/.goreleaser.yml:1]: this file should be placed at the root of the git repository

File: dist/config.yaml
Comment (mudler) [inline dist/config.yaml:1]: this directory is generated by running goreleaser, we shouldn't commit this to the repository

File: .goreleaser.yml
Comment (mudler) [inline .goreleaser.yml:38]: can you please check that this not collide with our current automation around release notes @nolanjacobson ? probably it is enough to set goreleaser to not generate release notes

**Comment by nolanjacobson** [comment]
@mudler  comments addressed.

File: .github/workflows/release.yaml
Comment (mudler) [inline .github/workflows/release.yaml:9]: I would suggest another workflow for this - the current `release` looks more a `deployment` workflow, not sure we want to mix triggers with this one

File: .github/workflows/release.yaml
Comment (mudler) [inline .github/workflows/release.yaml:112]:
```suggestion
          args: release --clean
```

File: .github/workflows/release.yaml
Comment (mudler) [inline .github/workflows/release.yaml:111]: let's try to use the latest version

Comment (mudler) [reply .github/workflows/release.yaml:111]:
```suggestion
          version: v2.1.0
```

**Review by mudler — APPROVED** [review]

**Comment by mudler** [comment]
@nolanjacobson the changelog settings don't seem to be valid at all ? https://goreleaser.com/customization/changelog/?h=changelog#changelog

**Comment by nolanjacobson** [comment]
Blocked by @j2d3  - John will merge once its ready.

**Pull request merged**
